### DISCUSSION TITLE
Using afterStateUpdated on dependent checkboxes in v2

### DISCUSSION THREAD

**Discussion by Rudston** [filamentphp/filament#9197] [Help] [answered]
Hello, I have a filament form builder form (version 2) displaying rows of data each containing two checkboxes: 'assignedItems' and 'optionalItems'. These fields are arrays.

If in any row, 'assigned' is unchecked y the user, then the 'optional' checkbox in the same row should also be unchecked.

I believe I can add a Closure function to the assigned items that will achieve this result.
Unfortunately this is not working... 

No errors in console / log

Please help!

Here is how each row is constructed:

`Grid::make(3)
                ->schema([
                    TextInput::make("items.$key")
                        ->label('Item')
                        ->extraAttributes(['class' => 'cursor-not-allowed'])
                        ->disableLabel()
                        ->disabled(),
                    Checkbox::make("assignedItems.$key")
                        ->extraAttributes(['class' => 'flex items-center'])
                        ->afterStateUpdated(function (Closure $set, $state) use ($key) {
                            if ($state === false) {
                                $set("optionalItems.$key", false);
                            }
                        })
                        ->label('Assigned'),
                    Checkbox::make("optionalItems.$key")
                        ->extraAttributes(['class' => 'flex items-center'])
                        ->label('Optional'),
                ]);`

**Comment by leandrocfe** [comment, marked as answer]
```php
Checkbox::make("assignedItems.$key")
->reactive()
...
```

**Reply by Rudston** [reply]
Thanks, works correctly now!
